### PR TITLE
a lighter version of pd-cap-recipes

### PR DESCRIPTION
**Pull request by alperkokmen**
- get rid of `Gemfile.lock` and use `gemspec` properly to manage gem dependencies.
- remove `hipchat` recipe. newest version of the gem (`1.4.0` at the time of writing this) offers the same features and more.
- remove `cap_gun` recipe.
- remove `comments` recipe.
- update `README` for the changes in this PR.
- refactor `git.cut_tag` into a method so we can call it from any task we like.
- bump version to `0.2.0` as there are many breaking changes.

### CODE REVIEW

**Comment by ranjib** [comment]
\m/

File: README.md [hunk 1]
```diff
@@ -9,52 +9,44 @@ These are various capistrano recipes used at [PagerDuty Inc.](http://www.pagerdu
...
-    end
-
-Then run 
-    bundle install
-    
+Add the following to your `Gemfile`.
+
+```ruby
+group :capistrano do
+  gem 'pd-cap-recipes', :git => 'git://github.com/PagerDuty/pd-cap-recipes.git'
```
Comment (esigler) [inline README.md]: Perhaps this should be published as a gem so path doesn't have to be set?

**Comment by esigler** [comment]
Also: woot!

File: README.md (hunk 1 above)
Comment (steverice) [inline README.md]: I'm :+1: for that, actually — even though I severely hope nobody else uses this gem.

Comment (dougbarth) [inline README.md]: We could throw a gem up on packagecloud.io if you don't want public consumption of the gem.

Comment (alperkokmen) [inline README.md]: @esigler @steverice sounds good. i will ping one of you guys regarding _how to publish a gem._

File: README.md [hunk 2]
```diff
@@ -9,52 +9,44 @@ These are various capistrano recipes used at [PagerDuty Inc.](http://www.pagerdu
...
+end
+```
+
+Then run
+
+    $ bundle install
```
Comment (steverice) [inline README.md]: :money_with_wings: :dollar: :moneybag:

File: lib/pd-cap-recipes/tasks/git.rb [hunk 3]
```diff
@@ -53,36 +59,11 @@ def remote_tag(tag)
...
     set :branch do
       return config[:_git_branch] if config[:_git_branch]
 
-      tag = config[:tag]
-      if !config[:tag]
-        tag = Capistrano::CLI.ui.ask green("Tag to deploy: ")
-        tag = tag.to_s.strip
-      end
-
+      # if tag is provided (e.g. -s tag=master-1234567890), use it. otherwise, cut a new tag.
```
Comment (steverice) [inline lib/pd-cap-recipes/tasks/git.rb]: :metal:

File: lib/pd-cap-recipes/tasks/git.rb [hunk 4]
```diff
@@ -108,6 +89,19 @@ def remote_tag(tag)
     end
   end
 
+  def git_cut_tag
+    repo = Grit::Repo.new('.')
+    raise 'You are currently in a detached head state. Cannot cut tag.' unless repo.head
+
+    new_tag = "#{repo.head.name}-#{Time.now.utc.to_i}"
+
+    git = GitRepo.new
```
Comment (steverice) [inline lib/pd-cap-recipes/tasks/git.rb]: `GitRepo` is part of `Grit`? Looks different from line 93, you sure both are not needed?

Comment (alperkokmen) [inline lib/pd-cap-recipes/tasks/git.rb]: `GitRepo` instance will forward everything to `Grit::Git` whereas `git` is an instance of `Grit::Repo.new('.')`.

Comment (steverice) [inline lib/pd-cap-recipes/tasks/git.rb]: Can't say that makes any sense, but :+1:

**Comment by steverice** [comment]
Get rid of that :hankey: 
trash

:ship:

**Pull request merged**